src/main/java/org/gelecekbilimde/scienceplatform/auth/controller/RoleApplicationController.java

```diff
@@ -0,0 +1,82 @@
+package org.gelecekbilimde.scienceplatform.auth.controller;
+
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.gelecekbilimde.scienceplatform.auth.model.RoleApplication;
+import org.gelecekbilimde.scienceplatform.auth.model.mapper.RoleApplicationToRoleApplicationResponseMapper;
+import org.gelecekbilimde.scienceplatform.auth.model.request.RoleChangeRequestsFilter;
+import org.gelecekbilimde.scienceplatform.auth.model.response.RoleApplicationsResponse;
+import org.gelecekbilimde.scienceplatform.auth.service.RoleApplicationService;
+import org.gelecekbilimde.scienceplatform.common.model.response.SuccessResponse;
+import org.hibernate.validator.constraints.UUID;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@Validated
+@RestController
+@RequestMapping("/api/v1")
+@RequiredArgsConstructor
+class RoleApplicationController {
+
+	private final RoleApplicationService roleApplicationService;
+
+
+	private final RoleApplicationToRoleApplicationResponseMapper roleApplicationToRoleApplicationResponseMapper = RoleApplicationToRoleApplicationResponseMapper.initialize();
```

Comment:
@coderabbitai The area here cannot be injected by spring. Because it's not a bean. In addition, it cannot inject because we use lombok anotations. Don't make this suggestion for mappers.